tests/lib/linter.js

```diff
@@ -1552,160 +1557,169 @@ describe("Linter", () => {
             });
-            linter.verify(code, config, filename, true);
+
+            linter.defineRule("checker", () => ({ Program: spy }));
+            linter.verify(code, config);
+            assert(spy.calledOnce);
         });
 
         it("variables should not be exported when in the node environment", () => {
             const code = "/* exported horse */\nvar horse = 'circus'";
-            const config = { rules: {}, env: { node: true } };
-
-            linter.reset();
-            linter.on("Program", () => {
+            const config = { rules: { checker: "error" }, env: { node: true } };
+            const spy = sandbox.spy(() => {
                 const scope = linter.getScope(),
                     horse = getVariable(scope, "horse");
 
                 assert.equal(horse, null); // there is no global scope at all
             });
-            linter.verify(code, config, filename, true);
+
+            linter.defineRule("checker", () => ({ Program: spy }));
+            linter.verify(code, config);
+            assert(spy.calledOnce);
         });
     });
 
     describe("when evaluating code containing a line comment", () => {
         const code = "//global a \n function f() {}";
 
         it("should not introduce a global variable", () => {
-
-            const config = { rules: {} };
-
-            linter.reset();
-            linter.on("Program", () => {
+            const config = { rules: { checker: "error" } };
+            const spy = sandbox.spy(() => {
                 const scope = linter.getScope();
 
                 assert.equal(getVariable(scope, "a"), null);
             });
-            linter.verify(code, config, filename, true);
+
+            linter.defineRule("checker", () => ({ Program: spy }));
+            linter.verify(code, config);
+            assert(spy.calledOnce);
         });
     });
 
     describe("when evaluating code containing normal block comments", () => {
         const code = "/**/  /*a*/  /*b:true*/  /*foo c:false*/";
 
         it("should not introduce a global variable", () => {
-            const config = { rules: {} };
-
-            linter.reset();
-            linter.on("Program", () => {
+            const config = { rules: { checker: "error" } };
+            const spy = sandbox.spy(() => {
                 const scope = linter.getScope();
 
                 assert.equal(getVariable(scope, "a"), null);
                 assert.equal(getVariable(scope, "b"), null);
                 assert.equal(getVariable(scope, "foo"), null);
                 assert.equal(getVariable(scope, "c"), null);
             });
-            linter.verify(code, config, filename, true);
+
+            linter.defineRule("checker", () => ({ Program: spy }));
+            linter.verify(code, config);
+            assert(spy.calledOnce);
         });
     });
 
     describe("when evaluating any code", () => {
-        const code = "";
+        const code = "x";
```

Comment:
All of the tests in this suite had previously been broken, because `Linter` doesn't run any rules when the source text is empty, so the functions with the assertions weren't getting called.